src/lib/viewers/media/DashViewer.js

```diff
@@ -488,17 +486,68 @@ class DashViewer extends VideoBaseViewer {
             textCues.push(new shaka.text.Cue(start, end, text));
         });
 
+        return textCues;
+    }
+
+    /**
+     * Loads auto-generated captions
+     *
+     * @public
+     * @param {Object} [skillsData] - optional skills data used to generate captions
+     * @return {void}
+     */
+    loadAutoGeneratedCaptions(skillsData) {
+        const skillsCards = skillsData || getProp(this.options, 'file.metadata.global.boxSkillsCards.cards');
+        // Do nothing if we don't have skills data or already have captions
+        if (!skillsCards || this.autoCaptionDisplayer) {
+            return;
+        }
+
+        // Save passed in skills data to avoid regenerating captions unnecessarily
+        this.skillsData = skillsData;
+
+        const textCues = this.parseTextCues(skillsCards);
+
         if (textCues.length > 0) {
             this.autoCaptionDisplayer = new shaka.text.SimpleTextDisplayer(this.mediaEl);
             this.autoCaptionDisplayer.append(textCues);
             this.player.configure({ textDisplayFactory: this.autoCaptionDisplayer });
+            // Update the subtitles/caption button to reflect auto-translation
+            this.mediaControls.setLabel(this.mediaControls.subtitlesButtonEl, __('media_auto_generated_captions'));
             this.mediaControls.initSubtitles(
                 [__('auto_generated')],
                 getLanguageName(this.options.location.locale.substring(0, 2))
             );
+        }
+    }
 
-            // Update the subtitles/caption button to reflect auto-translation
-            this.mediaControls.setLabel(this.mediaControls.subtitlesButtonEl, __('media_auto_generated_captions'));
+    /**
+     * Edits auto-generated captions
+     *
+     * @public
+     * @param {Object} skillsData - updated skills data
+     * @return {void}
+     */
+    editAutoGeneratedCaptions(skillsData) {
```

Comment:
It looks like there's some duplication between this function and `loadAutoGeneratedCaptions`. Should there be a third method that both of these call? I also think `update` might be more clear than `edit`, since no explicit user interaction is involved.

Reply:
Yep, will clean up in the next commit. Agreed that's a better name.